src/features/jserrors/instrument/index.js

```diff
@@ -22,76 +21,66 @@ export class Instrument extends InstrumentBase {
-      if (state.handleErrors) state.skipNext += 1
+    thisInstrument.ee.on('fn-start', function (args, obj, methodName) {
+      if (thisInstrument.abortHandler) thisInstrument.skipNext += 1
     })
 
-    state.ee.on('fn-err', function (args, obj, err) {
-      if (state.handleErrors && !err[NR_ERR_PROP]) {
+    thisInstrument.ee.on('fn-err', function (args, obj, err) {
+      if (thisInstrument.abortHandler && !err[NR_ERR_PROP]) {
         getOrSet(err, NR_ERR_PROP, function getVal() {
           return true
         })
         this.thrown = true
-        notice(err, undefined, state.ee)
+        notice(err, undefined, thisInstrument.ee)
       }
     })
 
-    state.ee.on('fn-end', function () {
-      if (!state.handleErrors) return
-      if (!this.thrown && state.skipNext > 0) state.skipNext -= 1
+    thisInstrument.ee.on('fn-end', function () {
+      if (!thisInstrument.abortHandler) return
+      if (!this.thrown && thisInstrument.skipNext > 0) thisInstrument.skipNext -= 1
     })
 
-    state.ee.on('internal-error', (e) => {
-      handle('ierr', [e, now(), true], undefined, FEATURE_NAMES.jserrors, state.ee)
+    thisInstrument.ee.on('internal-error', function (e) {
+      handle('ierr', [e, now(), true], undefined, FEATURE_NAMES.jserrors, thisInstrument.ee)
     })
 
-    const prevOnError = globalScope?.onerror
+    // Tack on our error handler onto the existing global one.
     globalScope.onerror = (...args) => {
-      if (prevOnError) prevOnError(...args)
+      if (this.origOnerror) this.origOnerror(...args)
       this.onerrorHandler(...args)
       return false
     }
 
-    try {
-      globalScope?.addEventListener('unhandledrejection', (e) => {
-        /** rejections can contain data of any type -- this is an effort to keep the message human readable */
-        const err = castReasonToError(e.reason)
-        handle('err', [err, now(), false, { unhandledPromiseRejection: 1 }], undefined, FEATURE_NAMES.jserrors, this.ee)
-      })
-    } catch (err) {
-      // do nothing -- addEventListener is not supported
-    }
-
-    try {
-      throw new Error()
-    } catch (e) {
-      // Only wrap stuff if try/catch gives us useful data. It doesn't in IE < 10.
-      if ('stack' in e) {
-        wrapTimer(this.ee)
-        wrapRaf(this.ee)
-
-        if ('addEventListener' in globalScope) {
-          wrapEvents(this.ee)
-        }
+    globalScope.addEventListener('unhandledrejection', (e) => {
+      /** rejections can contain data of any type -- this is an effort to keep the message human readable */
+      const err = castReasonToError(e.reason)
+      handle('err', [err, now(), false, { unhandledPromiseRejection: 1 }], undefined, FEATURE_NAMES.jserrors, this.ee)
+    }, {signal: this.removeOnAbort?.signal});
 
-        if (agentRuntime.xhrWrappable) {
-          wrapXhr(this.ee)
-        }
+    wrapRaf(this.ee);
+    wrapTimer(this.ee);
+    wrapEvents(this.ee);
+    wrapXhr(this.ee);
```

Comment:
Do we still need to check `.xhrWrappable` here?  Its being checked in the other features still

Reply:
Yep probably should. Unlike you're ok with just removing `xhrWrappable` altogether. v1220 integration issues has made me maybe too fearful of assuming givens. 

How likely do you think the chances are someone modified their global `XMLHttpRequest.prototype` such that `addEventListener` is no longer a function? (I think the logic isn't specifically about addEventListener but just using it as a way to see if it's been tampered with at all.)

Reply:
Impossible to know, i would leave the check in, mostly for "do no harm"